orchestrator/services/subscriptions.py

```diff
@@ -592,8 +596,11 @@ def build_extended_domain_model(subscription_model: SubscriptionModel) -> dict:
     def inject_in_use_by_ids(path_to_block: str) -> None:
         if not (in_use_by_subs := getattr_in(subscription_model, f"{path_to_block}.in_use_by")):
             return
-        block_instance_ids = [obj.in_use_by_id for obj in in_use_by_subs.col]
-        update_in(subscription, f"{path_to_block}.in_use_by_ids", block_instance_ids)
+
+        in_use_by_ids = [obj.in_use_by_id for obj in in_use_by_subs.col]
+        in_use_by_relations = [convert_subscription_instance(instance) for instance in in_use_by_subs]
+        update_in(subscription, f"{path_to_block}.in_use_by_ids", in_use_by_ids)
+        update_in(subscription, f"{path_to_block}.in_use_by_relations", in_use_by_relations)
```

Comment:
I hope someday we can build up an (extended) domain model from the ground using immutable datastructures instead of injecting stuff as an afterthought. Not easy to fix right now.